### ISSUE TITLE
Devices go offline after updating integration to 2024.5.2

### ISSUE THREAD

**Issue by Mis1ify** [make-all/tuya-local#1945]
**Describe the bug**
After updating HASSIO from 2024.04.04 to 2024.05.05 and updating tuya local to 2024.5.2 all of my added devices (except my smart plug) are going offline after reboot. I've even rebooted the host os and turned off the fuses to reboot switches.
**To Reproduce**
Steps to reproduce the behavior including the device the issue was observed with:
1. Update home assistant to may version (2024.5.5)
2. Update tuya local integration to may version (2024.5.2)
3. After 10 minutes or so all of my switches go offline.
**Expected behavior**
Devices shouldn't be unavailable.
**Additional context**
Before I've been just using an automation to reload the device, but now, homeassistant.reload_config_entry outputs an error:
`The config entry [Salon] Światła (tuya_local) with entry_id X cannot be unloaded because it is not in a recoverable state (ConfigEntryState.FAILED_UNLOAD)`

Please check if there are messages from Tuya Local in the Home
Assistant log (Settings / System / Logs), and include those too.

```
Rejestrator: homeassistant.config_entries
Źródło: config_entries.py:792
Pierwsze zdarzenie: 08:34:22 (6 zdarzenia)
Ostatnio zalogowany: 08:34:22

Error unloading entry [Salon] Światła for tuya_local
Error unloading entry [Fabo] Ledy pod biurkiem for tuya_local
Error unloading entry [Fabo] Światła for tuya_local
Error unloading entry [Przedpokój] Światła for tuya_local
Error unloading entry Wentylator/Ledy for tuya_local
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 792, in async_unload
    result = await component.async_unload_entry(hass, self)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/tuya_local/__init__.py", line 517, in async_unload_entry
    await async_delete_device(hass, config)
  File "/config/custom_components/tuya_local/device.py", line 655, in async_delete_device
    await hass.data[DOMAIN][device_id]["device"].async_stop()
  File "/config/custom_components/tuya_local/device.py", line 170, in async_stop
    await self._refresh_task
TypeError: object Future can't be used in 'await' expression
```
```
Rejestrator: homeassistant.components.number
Źródło: helpers/entity_platform.py:910
integracja: Liczba ([dokumentacja](https://www.home-assistant.io/integrations/number), [Problemy](https://github.com/home-assistant/core/issues?q=is%3Aissue+is%3Aopen+label%3A%22integration%3A+number%22))
Pierwsze zdarzenie: 08:34:22 (5 zdarzenia)
Ostatnio zalogowany: 08:34:22

Error while removing entity number.kuchnia_swiatla_timer_2
Error while removing entity number.salon_swiatla_timer_2
Error while removing entity number.fabo_swiatla_timer_switch_3
Error while removing entity number.przedpokoj_swiatla_timer_2
Error while removing entity number.wentylator_ledy_timer_2
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1378, in async_remove
    await self.__async_remove_impl(force_remove)
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1394, in __async_remove_impl
    await self.async_will_remove_from_hass()
  File "/config/custom_components/tuya_local/helpers/mixin.py", line 111, in async_will_remove_from_hass
    await self._device.async_unregister_entity(self)
  File "/config/custom_components/tuya_local/device.py", line 195, in async_unregister_entity
    await self.async_stop()
  File "/config/custom_components/tuya_local/device.py", line 170, in async_stop
    await self._refresh_task
TypeError: object Future can't be used in 'await' expression

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 910, in async_reset
    await entity.async_remove()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1383, in async_remove
    self.__remove_future.set_result(None)
asyncio.exceptions.InvalidStateError: invalid state

```

**Comment by make-all** [comment]
Maybe remove your automation?

There was an issue with 2024.4.0 and earlier of this integration with HA 2024.5, for which your automation was a workaround,. 
 That issue with startup was fixed in 2024.5.2, but there is a side effect of that fix when the integration is unloaded, which may be the cause of this problem on the first reload.

**Comment by Mis1ify** [comment]
Thanks for replying. I've changed the code in devices.py as pointed in #1888 and it works as a workaround. I haven't changed anything else.

**Comment by make-all** [comment]
OK, seems to be a duplicate of #1888 then, which will be fixed by the #1917 changes in next release
